File: WinPaletter/Store/Store_Intro.vb
Imports System.ComponentModel

Public Class Store_Intro
    Private Sub Store_Intro_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        CheckBox1.Checked = My.Settings.Store.ShowTips
        LoadLanguage
        ApplyStyle(Me)
        Icon = Store.Icon
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If TablessControl1.SelectedIndex + 1 <= TablessControl1.TabPages.Count - 1 Then TablessControl1.SelectedIndex += 1

        If Button1.Text = My.Lang.Finish Then
            Close()
            TablessControl1.SelectedIndex = 0
            Button1.Text = My.Lang.Next
        End If

        If TablessControl1.SelectedIndex = TablessControl1.TabPages.Count - 1 Then
            Button1.Text = My.Lang.Finish
            CheckBox1.Visible = True
        Else
            Button1.Text = My.Lang.Next
            CheckBox1.Visible = False
        End If
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        If TablessControl1.SelectedIndex - 1 >= 0 Then TablessControl1.SelectedIndex -= 1
        Button1.Text = My.Lang.Next
        CheckBox1.Visible = False
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Process.Start(My.Resources.Link_StoreOnlineSourceCreation)
    End Sub

    Private Sub Store_Intro_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        My.Settings.Store.ShowTips = CheckBox1.Checked
        My.Settings.Store.Save()
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Process.Start(My.Resources.Link_StoreUpload)
    End Sub

    Private Sub Button5_Click(sender As Object, e As EventArgs) Handles Button5.Click
        Process.Start(My.Resources.Link_StoreSourcesExtension)
    End Sub

    Private Sub Form_HelpButtonClicked(sender As Object, e As CancelEventArgs) Handles Me.HelpButtonClicked
        Process.Start(My.Resources.Link_Wiki & "/WinPaletter-Store-basics")
    End Sub
End Class

File: WinPaletter/Store/Store_Intro.cs
﻿using System;
using System.ComponentModel;
using System.Diagnostics;
using System.Windows.Forms;

namespace WinPaletter
{
    /// <summary>
    /// Introduction form for the store.
    /// </summary>
    public partial class Store_Intro
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="Store_Intro"/> class.
        /// </summary>
        public Store_Intro()
        {
            InitializeComponent();
        }
        private void Store_Intro_Load(object sender, EventArgs e)
        {
            CheckBox1.Checked = Program.Settings.Store.ShowTips;
            this.LoadLanguage();
            ApplyStyle(this);
            Icon = FormsExtensions.Icon<Store>();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (TablessControl1.SelectedIndex + 1 <= TablessControl1.TabPages.Count - 1)
                TablessControl1.SelectedIndex += 1;

            if ((Button1.Text ?? string.Empty) == (Program.Lang.Strings.General.Finish ?? string.Empty))
            {
                Close();
                TablessControl1.SelectedIndex = 0;
                Button1.Text = Program.Lang.Strings.General.Next;
            }

            if (TablessControl1.SelectedIndex == TablessControl1.TabPages.Count - 1)
            {
                Button1.Text = Program.Lang.Strings.General.Finish;
                CheckBox1.Visible = true;
            }
            else
            {
                Button1.Text = Program.Lang.Strings.General.Next;
                CheckBox1.Visible = false;
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            if (TablessControl1.SelectedIndex - 1 >= 0)
                TablessControl1.SelectedIndex -= 1;
            Button1.Text = Program.Lang.Strings.General.Next;
            CheckBox1.Visible = false;
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            Process.Start(Links.Wiki.StoreCreateSource);
        }

        private void Store_Intro_FormClosing(object sender, FormClosingEventArgs e)
        {
            Program.Settings.Store.ShowTips = CheckBox1.Checked;
            Program.Settings.Store.Save();
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            Process.Start(Links.Wiki.StoreUpload);
        }

        private void Button5_Click(object sender, EventArgs e)
        {
            Process.Start(Links.Wiki.StoreExtension);
        }

        private void Form_HelpButtonClicked(object sender, CancelEventArgs e)
        {
            Process.Start(Links.Wiki.StoreBasics);
        }
    }
}